src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -18,27 +18,41 @@
  */
 
 import React from 'react';
+import { i18n } from '@kbn/i18n';
 import { EuiImage } from '@elastic/eui';
-import { ChromeNavLink, CoreStart } from '../../../';
+import { AppCategory } from 'src/core/types';
+import { ChromeNavLink, CoreStart, ChromeRecentlyAccessedHistoryItem } from '../../../';
 import { HttpStart } from '../../../http';
 
-function isModifiedEvent(event: MouseEvent) {
+function isModifiedEvent(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
   return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
 }
 
 function LinkIcon({ url }: { url: string }) {
   return <EuiImage size="s" alt="" aria-hidden={true} url={url} />;
 }
 
-export type NavLink = ReturnType<typeof euiNavLink>;
+export interface NavLink {
```

Comment:
`NavLinkThatIsActuallyReadyToBePassedIntoEui`? 😬 

I think created this type and function last time I was touching the nav to try to clean up the situation a bit but I may simply added to the competing standard.

There's currently `ChromeNavLink`, `RecentNavLink`, `ChromeRecentlyAccessedHistoryItem`, `LegacyNavLink`... They all are similar but none are quite ready to be just passed into EUI and none are completely interoperable.

There's probably some tech debt cleanup work that either Core UI or Platform needs to take on to clean up some of the old chrome work because there are some really gnarly corners.

Added a short comment that maybe helps (just indicating that these links are ready for EUI)